{bio}[foss/2018a] Augustus 3.3 and dependencies

Serpens opened the pull request:
Augustus update to version 3.3 and the FOSS 2018a toolchain.

Dependencies:
- lpsolve 5.5.2.5
- BamTools 2.5.1

Serpens commented:
All tests passed now

verdurin commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3-foss-2018a.eb:
```diff
@@ -0,0 +1,40 @@
+toolchain = {'name': 'foss', 'version': '2018a'}
+
+sources = ['augustus-%(version)s.tar.gz']
+source_urls = ['http://bioinf.uni-greifswald.de/augustus/binaries/']
```
Please add a checksum with `--inject-checksums`.

verdurin commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3_bamtools_includepath.patch:
```diff
@@ -0,0 +1,25 @@
+diff -Nru augustus.orig/auxprogs/bam2hints/Makefile augustus/auxprogs/bam2hints/Makefile
```
Please include a comment at the top of the patch indicating the reason for including it.

verdurin commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3-foss-2018a.eb:
```diff
@@ -0,0 +1,40 @@
+easyblock = 'MakeCp'
```
@Serpens could you explain why you have changed the `easyblock` here? Is it an upstream change?

Serpens replied:
I changed it because I started writing this easyconfig before I noticed the older one and started from scratch. I can change to `ConfigureMake`, but the `configure` step is skipped anyway, so I'm not sure it makes any difference.

verdurin commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3-foss-2018a.eb:
```diff
@@ -0,0 +1,40 @@
+    ('SQLite', '3.21.0'),
+]
+
+files_to_copy = ['bin', 'include']
```
Is there a specific reason why the `config` and `scripts` directories are no longer being included?

Serpens replied:
No reason. I'll add them

verdurin commented on easybuild/easyconfigs/l/lpsolve/lpsolve-5.5.2.5-foss-2018a.eb:
```diff
@@ -0,0 +1,32 @@
+toolchain = {'name': 'foss', 'version': '2018a'}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['lp_solve_%(version)s_source.tar.gz']
```
Please add a checksum here too in the same way.

boegel replied:
@Serpens Checksum here is still missing?

Please include `SHA256` checksums for all new easyconfigs, we'll start enforcing this soon (see #5005)

Serpens replied:
I added it

Serpens commented:
@verdurin I've added fixes you requested. `include` directory is not installed by `make install`, so I guess it's not needed after compilation. I removed it from sanity check.

boegel commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3-foss-2018a.eb:
```diff
@@ -0,0 +1,45 @@
+              'bin/fastBlockSearch', 'bin/filterBam', 'bin/getSeq', 'bin/homGeneMapping', 'bin/joingenes',
+              'bin/load2sqlitedb', 'bin/prepareAlign'],
+    'dirs': ['config', 'scripts'],
+}
```
@Serpens missing compared to existing `AUGUSTUS` easyconfig:

```python
modextrapaths = {'PATH': 'scripts'}

modextravars = {'AUGUSTUS_CONFIG_PATH': '%(installdir)s/config'}
```

boegel commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3-foss-2018a.eb:
```diff
@@ -0,0 +1,45 @@
+
+skipsteps = ['configure']
+
+buildopts = 'COMPGENEPRED=true SQLITE=true ZIPINPUT=true CXX="$CXX" LINK.cc="$CXX"'
```
@Serpens Please also run `make clean`, apparently the tarball includes prebuilt binaries, cfr. #6049:

```python
prebuildopts = "make clean && "
```

Serpens replied:
After `make clean` there's no `bin/bam2wig`

boegel commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3_config.patch:
```diff
@@ -0,0 +1,17 @@
+diff -Nru augustus.orig/common.mk augustus/common.mk
```
@Serpens Comment + author is still missing here?

Serpens replied:
This patch is no longer needed after adding `buildopts`

boegel commented on easybuild/easyconfigs/a/AUGUSTUS/AUGUSTUS-3.3_no_symlinks_install.patch:
```diff
@@ -0,0 +1,25 @@
+don't try to link binaries in /usr/local/bin
```
@Serpens Please also mention author or upstream reference to source of this patch

Serpens replied:
Sorry about that

boegel commented on easybuild/easyconfigs/b/BamTools/BamTools-2.5.1-foss-2018a.eb:
```diff
@@ -0,0 +1,19 @@
+
+checksums = ['4abd76cbe1ca89d51abc26bf43a92359e5677f34a8258b901a01f38c897873fc']
+
+builddependencies = [('CMake', '3.9.5')]
```
@Serpens Please bump `CMake` version to `3.10.2` (we use this version in all `2018a` easyconfigs currently in `develop`

Serpens commented:
@boegel done

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2070.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/f02192735a30314665b7a1be97bf6aa3 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2473.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/4fa2c05f2b8ae1b394fe0ff5db3d515c for a full test report.

boegel commented:
Going in, thanks @Serpens!
